21 some doc feedback (#25)
capitalone/rubicon: README.md

[![Publish Package](https://github.com/capitalone/rubicon/actions/workflows/publish-package.yml/badge.svg)](https://github.com/capitalone/rubicon/actions/workflows/publish-package.yml)
[![Publish Docs](https://github.com/capitalone/rubicon/actions/workflows/publish-docs.yml/badge.svg)](https://github.com/capitalone/rubicon/actions/workflows/publish-docs.yml)

## Purpose

Rubicon is a data science tool that captures and stores model training and
execution information, like parameters and outcomes, in a repeatable and

> searchable way. Rubicon's git integration associates these inputs and outputs

Reviewer: ```suggestion
searchable way. Rubicon's `git` integration associates these inputs and outputs
```

directly with the model code that produced them to ensure full auditability and

> reproducibility for both developers and stakeholders alike. And the Rubicon
> dashboard makes it easy to explore, filter, visualize, and share recorded work.

Reviewer: ```suggestion
reproducibility for both developers and stakeholders alike. While experimenting,
the Rubicon dashboard makes it easy to explore, filter, visualize, and share
recorded work.
```

---

## Components

Rubicon is composed of three parts:

* A Python library for storing and retrieving model inputs, outputs, and
  analyses to filesystems that’s powered by

>   [fsspec](https://filesystem-spec.readthedocs.io/en/latest/?badge=latest)

Reviewer: ```suggestion
  [`fsspec`](https://filesystem-spec.readthedocs.io/en/latest/?badge=latest)
```

* A dashboard for exploring, comparing, and visualizing logged data built with

>   [dash](https://dash.plotly.com/)

Reviewer: ```suggestion
  [`dash`](https://dash.plotly.com/)
```

> * And a process for sharing a selected subset of logged data with collaborators

Reviewer: ```suggestion
* A process for sharing a selected subset of logged data with collaborators
```

Author: i think it reads better the way it is here

>   or reviewers that leverages [intake](https://intake.readthedocs.io/en/latest/)

Reviewer: ```suggestion
  or reviewers that leverages [`intake`](https://intake.readthedocs.io/en/latest/)
```

> ## Workflow

Reviewer: maybe "Workflow" should be the first subsection of "Use" before the example to make it flow a little better

Author: i don't want the example to look intimidating, which it might if it starts with a couple of paragraphs -- but maybe we can switch this later based on more feedback

> Use the Rubicon library to capture model inputs and outputs overtime. It can be

Reviewer: ```suggestion
Use the Rubicon library to capture model inputs and outputs over time. It can be
```

easily integrated into existing Python models or pipelines and supports both
concurrent logging (so multiple experiments can be logged in parallel) and
asynchronous communication with S3 (so network reads and writes won’t block).

Meanwhile, periodically review the logged data within the Rubicon dashboard to
steer the model tweaking process in the right direction. The dashboard lets you
quickly spot trends by exploring and filtering your logged results and
visualizes how the model inputs impacted the model outputs.

After the model has met its goals, use the collobaration process to share the
relevant subsets of logged data with model reviewers or stakeholders, giving

> everyone the context necessary for a peace of mind model review and approval.

Reviewer: this sentence reads really awkwardly

## Use

Here's a simple example:

```python
from rubicon import Rubicon

rubicon = Rubicon(
    persistence="filesystem", root_dir="/rubicon-root", auto_git_enabled=True
)

project = rubicon.create_project(
    "Hello World", description="Using rubicon to track model results over time."
)

experiment = project.log_experiment(

>     training_metadata=[SklearnTrainingMetadata("sklearn.datasets", "my-data-set")],

Reviewer: what is `SklearnTrainingMetadata`?

> accuracy = rfc.score(X_test, y_test)

Reviewer: what is `rfc`?
